### PR TITLE
Fix reward model evaluation

### LINKED ISSUE #397: Get model evaluation working on the reward model trainer
Based on #313 , we are having issues with model evaluation in the reward model trainer (code in model/ranking). It seems that the evaluation results are not being computed and logged for some reason. 

Preliminary research ~mentioned in this post https://github.com/LAION-AI/Open-Assistant/pull/313#issuecomment-1370430372, to summarize it may have something to do with the "evaluation_strategy" argument in TrainerArgs.~ was incorrect, see comment below.

It is extremely important that we have a normalized and robust test suite for our models in order to help select between them for large scale training. I've discussed this issue with @huyouare on discord and he seems to want to take it over. Will leave it to him to intro himself.

pinging @theblackcat102

### PR DESCRIPTION
**Pull request by maw501**
Closes #397. A few other minor tidy-up pieces too.

### CODE REVIEW

File: model/reward/instructor/trainer.py [hunk 1]
```diff
@@ -178,11 +183,11 @@ def prediction_step(
         args=args,
         loss_function=training_conf["loss"],
         train_dataset=train,
-        eval_dataset=evals,
+        eval_dataset=torch.utils.data.ConcatDataset(evals.values()),
```
Comment (sanagno) [inline model/reward/instructor/trainer.py:186]: do we actually want this?

Comment (maw501) [reply model/reward/instructor/trainer.py:186]: A `KeyError` is raised if not, since `evals` is a `dict`, not a `Dataset`. I'm not sure why it was created like this in [`get_datasets`](https://github.com/LAION-AI/Open-Assistant/blob/4250425c94b550df8b9b90082d52923525cbcb7e/model/reward/instructor/utils.py#L121) - I could amend it there?

Comment (sanagno) [reply model/reward/instructor/trainer.py:186]: Aha I see, it is created as a dict to follow the sft training I guess, but perhaps it needs some special handling to compute the losses. Maybe we can change this in the future.

**Review by sanagno — APPROVED** [review]
Looks good, @theblackcat102 if you want to take a look
